internal/conf/config.go

```diff
@@ -10,16 +10,18 @@ import (
 
 // DefaultOperatorConfig holds the default values of OperatorConfig.
 var DefaultOperatorConfig = OperatorConfig{
-	SmbdContainerImage:     "quay.io/samba.org/samba-server:latest",
-	SvcWatchContainerImage: "quay.io/samba.org/svcwatch:latest",
-	SmbdContainerName:      "samba",
-	WinbindContainerName:   "wb",
-	WorkingNamespace:       "",
-	SambaDebugLevel:        "",
-	StatePVCSize:           "1Gi",
-	ClusterSupport:         "",
-	SmbServicePort:         445,
-	SmbdPort:               445,
+	SmbdContainerImage:        "quay.io/samba.org/samba-server:latest",
+	SmbdMetricsContainerImage: "quay.io/samba.org/samba-metrics:latest",
+	SvcWatchContainerImage:    "quay.io/samba.org/svcwatch:latest",
+	SmbdContainerName:         "samba",
+	WinbindContainerName:      "wb",
+	WorkingNamespace:          "",
+	SambaDebugLevel:           "",
+	StatePVCSize:              "1Gi",
+	ClusterSupport:            "",
+	SmbServicePort:            445,
+	SmbdPort:                  445,
+	WithMetricsExporter:       false,
```

Comment:
I have a general issue with booleans when used in in certain contexts like configuration or other kinds of persistent data. I'd like to suggest a tweak to this to take a string and tweak the name to something like "MetricsExporterMode" accept values such as empty string (""), "enabled", "disabled". Then in the future, if we need different approaches for the exporter, say "debug" hypothetically we don't get an explosion of fields like you would if you needed to keep adding various on/off booleans for features.  Basically, I want an "enum" but since we can't directly use an enum - we have an enumerated set of string values. If this isn't clear, feel free to ping me and I'll try to elaborate more.

It is acceptable to fail the operator if they feed in a unknown value.

Reply:
ACK. Fixed (+rebase over latest master)